code/game/machinery/iv_drip.dm

```diff
@@ -157,11 +168,34 @@
 		to_chat(user, span_notice("You attach [W] to [src]."))
 		user.log_message("attached a [W] to [src] at [AREACOORD(src)] containing ([reagent_container.reagents.get_reagent_log_string()])", LOG_ATTACK)
 		add_fingerprint(user)
-		update_appearance()
+		update_appearance(UPDATE_ICON)
 		return
 	else
 		return ..()
 
+/// Checks whether the IV drip transfer rate can be modified with AltClick
+/obj/machinery/iv_drip/proc/can_use_alt_click(mob/user)
+	if(!can_interact(user))
+		return FALSE
+	if(istype(src, /obj/machinery/iv_drip/plumbing)) // AltClick is used for rotation there
+		return FALSE
+	if(!attached)
+		return FALSE
+	if(!get_reagents())
+		return FALSE
```

Comment:
Don't forget

Reply:
@Mothblocks asked in https://github.com/tgstation/tgstation/pull/71403#discussion_r1027644503 to move this kind of checks into a separate proc, which is what I did here too. There is a new proc now - `can_use_alt_click` that contains all the checks and I call `..()` only once.

Reply:
Thats not the point! Its still just 4 if-statements split apart, if you read down on that comment they're also still asking you to merge the if-statements

Reply:
That is the point of guard clauses. Merged if will have worse readability, and it is the same in terms of computation. Separate if statements are easier to maintain. This is against my religion.

Reply:
I disagree but also don’t care enough to die on this specific hill

Reply:
I think it's fine for these to be separate if's, it was the copying and pasting of the bodies I took concern with

Reply:
Fair enough

Reply:
So can I leave it as is?